Assets/UnitState.cs

```diff
@@ -0,0 +1,14 @@
+﻿using UnityEngine;
+
+public class UnitState : MonoBehaviour {
```

Comment:
what does this do?

Reply:
This is not the best name to be fair, but it essentially just stores the source stronghold of the unit object. Used in the code above.